adapter-base/src/test/java/org/eclipse/hono/adapter/resourcelimits/PrometheusBasedResourceLimitChecksTest.java

```diff
@@ -557,6 +557,39 @@ public void testMessageLimitUsesValueFromCache(final VertxTestContext ctx) {
                 }));
     }
 
+    /**
+     * Verifies that the message limit is not exceeded if no value is in the cache yet and the
+     * Prometheus query is still running.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testMessageLimitFallsBackToDefaultValueIfQueryStillRunning(final VertxTestContext ctx) {
+
+        when(dataVolumeCache.get(anyString(), any(BiFunction.class)))
+            .then(invocation -> {
+                final CompletableFuture<LimitedResource<Long>> result = new CompletableFuture<>();
+                return result;
+            });
```

Comment:
It could be simplified as `.then(invocation -> new CompletableFuture<LimitedResource<Long>>());`

Reply:
Anything else?